### PR TITLE
REFACTOR: codes structure primitive

### PR DESCRIPTION
**Pull request by ring630**

### CODE REVIEW

File: src/pyedb/dotnet/edb_core/cell/primitive.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        super().__init__(pedb, edb_object)
+        self._app = self._pedb
+        self._core_stackup = pedb.stackup
+        self._core_net = pedb.nets
+        self.primitive_object = self._edb_object
+
+    @property
+    def type(self):
+        """Return the type of the primitive.
+        Allowed outputs are ``"Circle"``, ``"Rectangle"``,``"Polygon"``,``"Path"`` or ``"Bondwire"``.
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/cell/primitive.py:49]:
```suggestion
        
        Expected ouput is among ``"Circle"``, ``"Rectangle"``,``"Polygon"``,``"Path"`` or ``"Bondwire"``.
```

File: src/pyedb/dotnet/edb_core/cell/primitive.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        str
+        """
+        try:
+            return self._edb_object.GetPrimitiveType().ToString()
+        except AttributeError:  # pragma: no cover
+            return ""
+
+    @property
+    def net_name(self):
+        """Get or Set the primitive net name.
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/cell/primitive.py:62]:
```suggestion
        """Get the primitive net name.
```

File: src/pyedb/dotnet/edb_core/cell/primitive.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    def net_name(self, name):
+        if isinstance(name, str):
+            net = self._app.nets.nets[name].net_object
+            self.primitive_object.SetNet(net)
+        else:
+            try:
+                if isinstance(name, str):
+                    self.net = name
+                elif isinstance(name, NetDotNet):
+                    self.net = name.name
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/cell/primitive.py:80]:
```suggestion
                self.net = name.name
```
We don't need to test `if isinstance(name, str)` twice.

File: src/pyedb/dotnet/edb_core/cell/primitive.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        """Get the primitive edb layer object."""
+        try:
+            layer_name = self.primitive_object.GetLayer().GetName()
+            return self._pedb.stackup.layers[layer_name]
+        except (KeyError, AttributeError):  # pragma: no cover
+            return None
+
+    @property
+    def layer_name(self):
+        """Get or Set the primitive layer name.
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/cell/primitive.py:95]:
```suggestion
        """Get the primitive layer name.
```

**Review by svandenb-dev — APPROVED** [review]
LGTM

File: src/pyedb/dotnet/edb_core/cell/primitive.py [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        self._app = self._pedb
+        self._core_stackup = pedb.stackup
+        self._core_net = pedb.nets
+        self.primitive_object = self._edb_object
+
+    @property
+    def type(self):
+        """Return the type of the primitive.
+
+        Expected ouput is among ``"Circle"``, ``"Rectangle"``,``"Polygon"``,``"Path"`` or ``"Bondwire"``.
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/cell/primitive.py:51]:
```suggestion
        Expected output is among ``"Circle"``, ``"Rectangle"``,``"Polygon"``,``"Path"`` or ``"Bondwire"``.
```

**Comment by ring630** [comment]
> I left some minor changes. @ring630 Is this related to the architecture changes we wanted to perform to be closed to the EDB API ?



> I left some minor changes. @ring630 Is this related to the architecture changes we wanted to perform to be closed to the EDB API ?

@SMoraisAnsys Yes. I am moving EdbDotNet classes one by one out from donet folder. These classes were created by Massimo as a preparation for grpc migration. The legacy migration plan is dropped.

**Review by svandenb-dev — APPROVED** [review]
LGTM

**Pull request merged**